opencti-platform/opencti-graphql/src/modules/malwareAnalysis/malwareAnalysis.ts

```diff
@@ -0,0 +1,138 @@
+import type { StixMalwareAnalysis, StoreEntityMalwareAnalysis } from './malwareAnalysis-types';
+import { ENTITY_TYPE_MALWARE_ANALYSIS, INPUT_ANALYSIS_SCO, INPUT_HOST_VM, INPUT_INSTALLED_SOFTWARE, INPUT_OPERATING_SYSTEM, INPUT_SAMPLE } from './malwareAnalysis-types';
+import { ABSTRACT_STIX_DOMAIN_OBJECT } from '../../schema/general';
+import malwareAnalysisTypeDefs from './malwareAnalysis.graphql';
+import malwareAnalysisResolvers from './malwareAnalysis-resolvers';
+import { RELATION_ANALYSIS_OF, RELATION_CHARACTERIZES, RELATION_DYNAMIC_ANALYSIS_OF, RELATION_STATIC_ANALYSIS_OF } from '../../schema/stixCoreRelationship';
+import { ENTITY_TYPE_MALWARE } from '../../schema/stixDomainObject';
+import { REL_BUILT_IN } from '../../database/stix';
+import type { ModuleDefinition } from '../../schema/module';
+import { registerDefinition } from '../../schema/module';
+
+import convertMalwareAnalysisToStix from './malwareAnalysis-converter';
+import {
+  ENTITY_HASHED_OBSERVABLE_ARTIFACT,
+  ENTITY_HASHED_OBSERVABLE_STIX_FILE,
+  ENTITY_NETWORK_TRAFFIC,
+  ENTITY_SOFTWARE,
+  isStixCyberObservable
+} from '../../schema/stixCyberObservable';
+import {
+  ATTRIBUTE_ANALYSIS_SCO,
+  ATTRIBUTE_HOST_VM,
+  ATTRIBUTE_INSTALLED_SOFTWARE,
+  ATTRIBUTE_OPERATING_SYSTEM,
+  ATTRIBUTE_SAMPLE,
+  RELATION_ANALYSIS_SCO,
+  RELATION_HOST_VM,
+  RELATION_INSTALLED_SOFTWARE,
+  RELATION_OPERATING_SYSTEM,
+  RELATION_SAMPLE
+} from './malwareAnalysis-domain';
+
+const MALWARE_ANALYSIS_DEFINITION: ModuleDefinition<StoreEntityMalwareAnalysis, StixMalwareAnalysis> = {
+  type: {
+    id: 'malwareAnalyses',
+    name: ENTITY_TYPE_MALWARE_ANALYSIS,
+    category: ABSTRACT_STIX_DOMAIN_OBJECT,
+    aliased: false
+  },
+  graphql: {
+    schema: malwareAnalysisTypeDefs,
+    resolver: malwareAnalysisResolvers
+  },
+  identifier: {
+    definition: {
+      [ENTITY_TYPE_MALWARE_ANALYSIS]: [{ src: 'product' }, { src: 'result_name' }, { src: 'submitted' }]
```

Comment:
Why do we want submitted as part of identifier ?

Reply:
It's the date of the submission of the analysis. Sam says it's important and defines (when combined with the product and result_name) the Object